app/Main.hs

```diff
@@ -230,21 +261,31 @@ optsParser =
               ]
         )
     <*> configParser
-    <*> cabalDefaultExtensionsParser
+    <*> cabalOptsParser
     <*> sourceTypeParser
     <*> (many . strArgument . mconcat)
       [ metavar "FILE",
         help "Haskell source files to format or stdin (the default)"
       ]
 
-cabalDefaultExtensionsParser :: Parser CabalDefaultExtensionsOpts
-cabalDefaultExtensionsParser =
-  CabalDefaultExtensionsOpts
+cabalOptsParser :: Parser CabalOpts
+cabalOptsParser =
```

Comment:
IMO there is no good reason to take default extensions and language, but not the list of packages. I can imagine two distinct cases:

* There is no Cabal file or it cannot be found. Then the user is going to provide the extensions and other info on the command line.
* There is a Cabal file that we can find. Then take all the useful info from that file.

Let's have a single flag `--cabal` that does all at once like a originally suggested.

Reply:
I added a separate flag, because dependency detection should be enabled by default IMO, but extension detection isn't enabled by default at the moment (If I understand correctly the Main.hs file)

Reply:
The current `--cabal-default-extensions` flag tells Ormolu to try to look for a Cabal file to take default extensions from there. We want to keep giving the user this choice (whether or not they want Ormolu to look at Cabal files), but if a Cabal file is found we want to also take the list of direct dependencies from there. So, yes, it would take packages from the Cabal file by default but only if user wants Ormolu to look for Cabal files at all.

Reply:
I'm just afraid that people who do not use the `--cabal-default-extensions` flag at the moment (which will be renamed `--cabal`) will miss out the accuracy improvement for operator chains formatting

Reply:
I understand, but we should not try to look for Cabal files unconditionally at this point. What we can do is discuss whether we want `--cabal` to be enabled by default. We could also add `--no-cabal` to opt out from this. But let's tackle questions one by one and not try to do too much in a single PR.